packages/core/src/server.ts

```diff
@@ -602,13 +632,21 @@ export class Controller extends API {
         // Persist changes
         await Promise.all([this.storage.save(session), this.storage.save(acc)]);
 
-        // Add device to trusted devices
-        if (
-            this.context.device &&
-            !auth.trustedDevices.some(({ id }) => id === this.context.device!.id) &&
-            addTrustedDevice
-        ) {
-            auth.trustedDevices.push(this.context.device);
+        // Check if device isn't trusted
+        if (this.context.device && !auth.trustedDevices.some(({ id }) => id === this.context.device!.id)) {
+            // Add to trusted devices
+            if (addTrustedDevice) {
+                auth.trustedDevices.push(this.context.device);
+            }
+
+            // Send new login notification (it's a new or untrusted device)
+            if (acc.settings.notifications.newLogins) {
+                try {
+                    const location = this._buildLocationAndDeviceString(this.context.location, this.context.device);
+
+                    await this.messenger.send(acc.email, new NewLoginMessage({ location }));
+                } catch (e) {}
+            }
```

Comment:
I think we can/should do this asynchronously, since it's not critical if it fails and waiting for the message to be sent will add a significant delay.